pkg/pool-stable/test/meta/MetaStablePool.test.ts

```diff
@@ -583,104 +596,129 @@ describe('MetaStablePool', function () {
-            const { rate: newPriceRate0 } = await pool.instance.getPriceRateCache(tokens.first.address);
-            const { rate: newPriceRate1 } = await pool.instance.getPriceRateCache(tokens.second.address);
+            it('updates both caches', async () => {
+              const { rate: newPriceRate0 } = await pool.instance.getPriceRateCache(tokens.first.address);
+              const { rate: newPriceRate1 } = await pool.instance.getPriceRateCache(tokens.second.address);
 
-            expect(newPriceRate0).to.be.gt(oldPriceRate0);
-            expect(newPriceRate1).to.be.equal(oldPriceRate1);
+              expect(newPriceRate0).to.be.gt(oldPriceRate0);
+              expect(newPriceRate1).to.be.gt(oldPriceRate1);
 
-            const scalingFactors = await pool.instance.getScalingFactors();
-            expect(scalingFactors[0]).to.be.equal(scaleRate(newPriceRate0, tokens.first));
-            expect(scalingFactors[1]).to.be.equal(scaleRate(oldPriceRate1, tokens.second));
+              const scalingFactors = await pool.instance.getScalingFactors();
+              expect(scalingFactors[0]).to.be.equal(scaleRate(newPriceRate0, tokens.first));
+              expect(scalingFactors[1]).to.be.equal(scaleRate(newPriceRate1, tokens.second));
+            });
           });
         });
+      });
+
+      describe('setting', () => {
+        createPoolWithInitialRates(0);
+
+        sharedBeforeEach('grant role to admin', async () => {
+          const action = await actionId(pool.instance, 'setPriceRateCacheDuration');
+          await pool.vault.grantRole(action, admin);
+        });
 
-        context('after both caches expired', () => {
-          mockNewRatesAndAdvanceTime(cacheDurations[1] + 1);
+        const updatePriceRateCache = () => {
+          sharedBeforeEach('update price rate cache', async () => {
+            await pool.setPriceRateCacheDuration(tokens.first, MINUTE * 10, { from: admin });
+            await pool.setPriceRateCacheDuration(tokens.second, MINUTE * 10, { from: admin });
+          });
+        };
 
-          it('updates both caches', async () => {
-            const { rate: newPriceRate0 } = await pool.instance.getPriceRateCache(tokens.first.address);
-            const { rate: newPriceRate1 } = await pool.instance.getPriceRateCache(tokens.second.address);
+        context('when it is requested by the admin', () => {
+          context('when it did not passed the previous duration', () => {
+            mockNewRatesAndAdvanceTime(MINUTE / 2);
+            updatePriceRateCache();
+            itAdaptsTheScalingFactorsCorrectly();
+          });
 
-            expect(newPriceRate0).to.be.gt(oldPriceRate0);
-            expect(newPriceRate1).to.be.gt(oldPriceRate1);
+          context('when it passed the previous duration', () => {
+            mockNewRatesAndAdvanceTime(MINUTE * 2);
+            updatePriceRateCache();
+            itAdaptsTheScalingFactorsCorrectly();
+          });
+        });
```

Comment:
It seems like these tests don't really test the new duration is in effect?

Reply:
It does, note that `itAdaptsTheScalingFactorsCorrectly` is querying the provider directly and these where updated right before the assertion, then the current scaling factors are being tested against the latest values coming from the providers